qiskit/transpiler/passes/utils/check_cx_direction.py

```diff
@@ -47,9 +48,11 @@ def run(self, dag):
         self.property_set['is_direction_mapped'] = True
         edges = self.coupling_map.get_edges()
 
+        trivial_layout = Layout.generate_trivial_layout(*dag.qregs.values())
```

Comment:
I saw the WIP todo in the PR summary about this, but what is the functional difference between using a layout here versus building a qubit indices dict?

Reply:
Functionally, only that `Layout` is a bidirectional (instead of unidirectional) mapping from `Qubit`s to `int`s. Since I wrote that comment though, I've moved away from the idea of using `Layout`s as a generic mapping (outside the contexts of layout and routing) because the resulting code ends up for me harder to read.